src/mavsdk_server/src/plugins/calibration/calibration_service_impl.h

```diff
@@ -142,11 +142,10 @@ class CalibrationServiceImpl final : public rpc::calibration::CalibrationService
         register_stream_stop_promise(stream_closed_promise);
 
         auto is_finished = std::make_shared<bool>(false);
-
-        std::mutex subscribe_mutex{};
+        auto subscribe_mutex = std::make_shared<std::mutex>();
```

Comment:
I guess that works. Could you also use one `shared_ptr` around the mutex and bool together?

Reply:
The problem is that if passed as a reference, the reference inside the callback gets invalid once the function returns.

> Could you also use one shared_ptr around the mutex and bool together?

Not sure what you mean :thinking:. An `std::shared_ptr<std::pair<bool, std::mutex>>`?

Reply:
For instance, yes.

Reply:
Why is that? Not sure I get it :confused:

Reply:
Ah sorry, I just meant as an optimization, so you only need to do the reference counting once.

Reply:
Oh, I understand now! This function is called exactly once per subscription, and in the current state, that usually means at most once (because MAVSDK-C++ does not support registering more than one callback yet). So I don't really worry about the `shared_ptr` overhead.

But of course I can change it, I'll do that tomorrow :+1:

Reply:
Since it's tested, let's leave it!